src/app.js

```diff
@@ -1,79 +1,157 @@
 /* eslint-disable no-unused-vars */
 
-import {Utils} from '@natlibfi/melinda-commons';
+import ApiError, {Utils} from '@natlibfi/melinda-commons';
 import {amqpFactory, mongoFactory, logError, QUEUE_ITEM_STATE, OPERATIONS} from '@natlibfi/melinda-rest-api-commons';
-import {AMQP_URL, OPERATION, POLL_WAIT_TIME, MONGO_URI} from './config';
 import {promisify} from 'util';
-import {datastoreFactory} from './interfaces/datastore';
+import recordLoadFactory from './interfaces/datastore';
+import processFactory from './interfaces/processPoll';
+
+export default async function ({
+	amqpUrl, operation, pollWaitTime, mongoUri,
+	recordLoadApiKey, recordLoadLibrary, recordLoadUrl
+}) {
+	const setTimeoutPromise = promisify(setTimeout);
+	const {createLogger} = Utils;
+	const logger = createLogger(); // eslint-disable-line no-console
+	const OPERATION_TYPES = [OPERATIONS.CREATE, OPERATIONS.UPDATE];
+	const purgeQueues = false;
+	let amqpOperator;
+	let mongoOperator;
+	let recordLoadOperator;
+	let processOperator;
+
+	const app = await initApp();
+	// Soft shutdown function
+	app.on('close', async () => {
+		// Things that need soft shutdown
+		// amqp disconnect?
+		// mongo disconnect?
+	});
+
+	return app;
+
+	async function initApp() {
+		try {
+			amqpOperator = await amqpFactory(amqpUrl);
+			mongoOperator = await mongoFactory(mongoUri);
+			recordLoadOperator = recordLoadFactory(recordLoadApiKey, recordLoadLibrary, recordLoadUrl);
+			processOperator = processFactory(recordLoadApiKey, recordLoadLibrary, recordLoadUrl);
+			// Start loop
+			await checkProcess(true);
+		} catch (error) {
+			logError(error);
+			process.exit(1);
+		}
```

Comment:
I don't see the need to wrap these in try-catch and using let. If calling these functions fail the application will exit anyway and no custom error handler is needed.